### PR TITLE
spawn_test: prolong termination time to be more tolerant.

### PR DESCRIPTION
**Pull request by balusch**
pidfd_open(2) can fail on old kernels due to several reasons (e.g.
syscall not supported, O_NONBLOCK flags not supported), in which
case process:wait() falls back to waitpid(2) with backoff (at least
20ms), which may make it longer than 10ms.

Here we add the minimal backoff to 10ms so that the test can pass
on old kernels.

Signed-off-by: Jianyong Chen <baluschch@gmail.com>

### CODE REVIEW

**Comment by balusch** [comment]
@tchaikov Hi, kefu, I think I've found the reason.

My ArchLinux is hosted on a machine with kernel 5.4, which supports `pidfd_open(2)` but not the`PIDFD_NONBLOCK` flag(introduced in 5.10), and thus it returns `EINVAL` and `process::wait` falls back to use `waitpid(2)` which however returns 0(the state of the `sleep` process is not changed yet) and then after 20ms backoff it retries and finally succeeds.

**Comment by tchaikov** [comment]
@balusch hi Jianyong, thanks for your excellent analysis. that explains the symptom in your environment!

the test failures are not related to this change:

- clang++-15 + release + C++17: Seastar.unit.stall_detector, see [0]. tracked by #1485
- gcc-12 + debug + C++20: Seastar.unit.spawn, this is a known issue, tracked by #1320

---
[0]: `Seastar.unit.stall_detector`:
```
54/74 Test #54: Seastar.unit.stall_detector ...................***Failed    6.10 sec
[0/1] cd /home/circleci/project/build/release/tests/unit && /home/circleci/project/build/release/tests/unit/stall_detector_test -- -c 2
Running 5 test cases...
INFO  2023-02-09 04:15:22,535 seastar - Reactor backend: linux-aio
WARN  2023-02-09 04:15:22,535 [shard 0] seastar - Creation of perf_event based stall detector failed, falling back to posix timer: std::system_error (error system:13, perf_event_open() failed: Permission denied)
INFO  2023-02-09 04:15:22,536 [shard 0] seastar - Created fair group io-queue-0, capacity rate 2147483:2147483, limit 12582912, rate 16777216 (factor 1), threshold 2000
INFO  2023-02-09 04:15:22,536 [shard 0] seastar - IO queue uses 0.75ms latency goal for device 0
INFO  2023-02-09 04:15:22,536 [shard 0] seastar - Created io group dev(0), length limit 4194304:4194304, rate 2147483647:2147483647
INFO  2023-02-09 04:15:22,536 [shard 0] seastar - Created io queue dev(0) capacities: 512:2000:2000 1024:3000:3000 2048:5000:5000 4096:9000:9000 8192:17000:17000 16384:33000:33000 32768:65000:65000 65536:129000:129000 131072:257000:257000
WARN  2023-02-09 04:15:22,550 [shard 1] seastar - Creation of perf_event based stall detector failed, falling back to posix timer: std::system_error (error system:13, perf_event_open() failed: Permission denied)
random-seed=1797987249
INFO  2023-02-09 04:15:22,551 [shard 0] seastar - updated: blocked-reactor-notify-ms=10
/home/circleci/project/tests/unit/stall_detector_test.cc(92): fatal error: in "normal_case": critical check reports == 0 has failed [1 != 0]
INFO  2023-02-09 04:15:23,551 [shard 0] seastar - updated: blocked-reactor-notify-ms=25
INFO  2023-02-09 04:15:23,551 [shard 0] seastar - updated: blocked-reactor-notify-ms=10
INFO  2023-02-09 04:15:24,853 [shard 0] seastar - updated: blocked-reactor-notify-ms=25
INFO  2023-02-09 04:15:24,853 [shard 0] seastar - updated: blocked-reactor-notify-ms=10
INFO  2023-02-09 04:15:25,861 [shard 0] seastar - updated: blocked-reactor-notify-ms=25
INFO  2023-02-09 04:15:25,861 [shard 0] testlog - Starting spin test: userspace
```

File: tests/unit/spawn_test.cc [hunk 1]
```diff
@@ -129,10 +129,13 @@ SEASTAR_TEST_CASE(test_spawn_kill) {
             auto* wait_signaled = std::get_if<experimental::process::wait_signaled>(&wait_status);
             BOOST_REQUIRE(wait_signaled != nullptr);
             BOOST_CHECK_EQUAL(wait_signaled->terminating_signal, SIGTERM);
-            // sleep should be terminated in 10ms
             auto end = std::chrono::high_resolution_clock::now();
             auto ms = std::chrono::duration_cast<std::chrono::milliseconds>(end - start).count();
-            BOOST_CHECK_LE(ms, 10);
+            // sleep should be terminated in 10ms.
+            // pidfd_open(2) may fail and thus p.wait() falls back to
+            // waitpid(2) with backoff(at least 20ms).
```
Comment (tchaikov) [inline tests/unit/spawn_test.cc:136]: nit, please add a space before " (at".

File: tests/unit/spawn_test.cc [hunk 2]
```diff
@@ -129,10 +129,13 @@ SEASTAR_TEST_CASE(test_spawn_kill) {
...
+            // here we allow one backoff to be more tolerant.
```
Comment (tchaikov) [inline tests/unit/spawn_test.cc:137]:
```suggestion
            // to be more tolerant, the minimal backoff is added to 10ms, so the test
            // can pass on older kernels as well.
```

**Comment by tchaikov** [comment]
the spawn_test is a known one and is not fixed by this change. i was looking it, but no clues so far. probably i should drop it.

the stall_detector one might be a flaky test. let's note it down just in case.

File: tests/unit/spawn_test.cc (hunk 1 above)
Comment (balusch) [reply tests/unit/spawn_test.cc]: done

**Review by tchaikov — APPROVED** [review]
lgtm

**Comment by tchaikov** [comment]
@scylladb/seastar-maint could you help merge this one as well?

**Pull request merged**